python/ray/data/tests/test_ray_databricks.py

```diff
@@ -0,0 +1,145 @@
+import os.path
+
+import pytest
+import pyarrow as pa
+import pandas as pd
+from ray import cloudpickle as pickle
+import uuid
+import math
+import tempfile
+from collections import namedtuple
+from contextlib import contextmanager
+from unittest import mock
+import ray
+
+
+ChunkMeta = namedtuple('ChunkMeta', ['id', 'rowCount', 'byteCount'])
+
+
+@contextmanager
+def setup_mock(default_chunk_bytes):
+    """
+    `ray.data.from_spark` supports Databricks runtime, but it relies on databricks
+    internal APIs.
+    So in unit tests, we have to set up mocks for testing it.
+
+    Note that the mocked `persist_df_as_chunks` function accepts
+    pandas dataframe as a fake spark dataframe.
+    so in unit test we don't need to create real spark dataframe,
+    this simplifies unit testing code.
+    """
+    tmp_dir = tempfile.mkdtemp()
+
+    def persist_df_as_chunks(pandas_df, bytes_per_chunk):
+        arrow_tb = pa.Table.from_pandas(pandas_df)
+
+        total_nbytes = arrow_tb.arrow_tb
+        num_rows = len(pandas_df)
+
+        num_chunks = math.ceil(total_nbytes / bytes_per_chunk)
+        rows_per_chunk = num_rows // num_chunks
+
+        def gen_chunk(chunk_idx):
+            chunk_id = uuid.uuid4().hex
+
+            start_row_idx = chunk_idx * rows_per_chunk
+            end_row_idx = start_row_idx + rows_per_chunk
+
+            chunk_pdf = pandas_df[start_row_idx: end_row_idx]
+
+            chunk_table = pa.Table.from_pandas(chunk_pdf)
+
+            with open(os.path.join(tmp_dir, chunk_id), "wb") as f:
+                pickle.dump(chunk_table, f)
+
+            return ChunkMeta(
+                id=chunk_id,
+                rowCount=len(chunk_pdf),
+                byteCount=chunk_table.nbytes,
+            )
+        return [
+            gen_chunk(chunk_idx)
+            for chunk_idx in range(num_chunks)
+        ]
+
+    def read_chunk(chunk_id):
+        with open(os.path.join(tmp_dir, chunk_id), "rb") as f:
+            return pickle.load(f)
+
+    read_chunk_fn_path = os.path.join(tmp_dir, "read_chunk_fn.pkl")
+    with open(read_chunk_fn_path, "wb") as fp:
+        pickle.dump(read_chunk, fp)
+
+    MOCK_ENV = "_RAY_DATABRICKS_FROM_SPARK_READ_CHUNK_FN_PATH"
+
+    def unpersist_chunk(chunk_id):
+        os.remove(os.path.join(tmp_dir, chunk_id))
```

Comment:
we can remove this if we use tmp_patch fixture or a context manager.

Reply:
In unit tests it also tests that `unpersist_chunk` is called properly when `del ray_ds` is called, to check that these chunk files are removed after `del ray_ds`

Reply:
I think you can use `assert_called` if you want to make sure it's called.

Reply:
got it

Reply:
I'd check use `assert_called` and `tmp_dir` is cleaned up after `del ray_ds`.